consensus/abci.go

```diff
@@ -97,10 +105,11 @@ func (abci *TendermintABCI) InitChain(req tmtAbciTypes.RequestInitChain) tmtAbci
 // for the validators.
 //
 // Response:
-//		- Optional Key-Value tags for filtering and indexing
+// 		- Optional Key-Value tags for filtering and indexing
 func (abci *TendermintABCI) BeginBlock(req tmtAbciTypes.RequestBeginBlock) tmtAbciTypes.ResponseBeginBlock {
 	abci.logger.Debug("Beginning new block", "hash", req.Hash)
 	abci.db.UpdateBlockState(&req.Header)
+	abci.curBlockHeader = &req.Header
```

Comment:
Yes it has it right one line above the one you added :) 

```
abci.db.UpdateBlockState(&req.Header)
abci.curBlockHeader = &req.Header
```

So I think it would be a better design to encapsulate this responsibility into the blockState. So calling the `updateBlockState()` function would update the `proposerAddress` as well.